src/tile_layer.js

```diff
@@ -73,7 +72,7 @@ var TileLayer = L.TileLayer.extend({
         if (!templated) {
             return templated;
         } else {
-            return templated.replace('.png',
+            return templated.replace(/\.(png|jpg)/,
```

Comment:
Use the [stricter regex from GL](https://github.com/mapbox/mapbox-gl-js/blob/fe072b7f178345abe94fecd7ab3e4b85bc77018d/js/util/mapbox.js#L64). This regex will fail with a hypothetical map id of `pwn.jpg` for instance.